### PR TITLE
Hip monitoring (80x)

### PR DESCRIPTION
**Pull request by mtosi**
backport of PR https://github.com/cms-sw/cmssw/pull/15447
in addition I add the https://github.com/cms-sw/cmssw/pull/14900
because that is part of the monitoring needed for the hip
and it has not been merged yet

please, this is **urgent** for the hip monitoring
I hope there are no issues w/ the integration

he needed 3 new "keys" in the AlCaRecoTriggerBitsRcd payload
should be added by arun.kumar@cern.ch as discussed w/ @mmusich
as soon as the PR is integrated in 80x

the corresponding trigger paths should be integrated in the next days
as described in https://its.cern.ch/jira/browse/CMSHLT-981

I really hope (and somehow request) to have this PR integrated w/in few days
because we need this plots by next week in the GUI for the studies related to the hip
as requested by the HIP task force
@dmitrijus @vanbesien 

thanks much

### CODE REVIEW

**Comment by mtosi** [comment]
@dmitrijus @vanbesien @rovere @VinInn

**Comment by VinInn** [comment]
@cmsbuild , please test

File: DQM/TrackingMonitor/src/TrackingMonitor.cc [hunk 1]
```diff
@@ -110,7 +127,8 @@ TrackingMonitor::TrackingMonitor(const edm::ParameterSet& iConfig)
   if ( doPlotsVsBXlumi_ )
       theLumiDetails_ = new GetLumi( iConfig.getParameter<edm::ParameterSet>("BXlumiSetup"), c );
```
Comment (dmitrijus) [inline DQM/TrackingMonitor/src/TrackingMonitor.cc]: Make theLumiDetails_ a unique_ptr? I feel uncomfortable around unfreed object.

**Comment by dmitrijus** [comment]
+1

**Comment by mmusich** [comment]
@mtosi assuming this will be integrated soon in 80x, please confirm the update of the TriggerBit selection is needed only on the Prompt processing and not on Express (I am assuming you'd like to have it run on the PDs mentioned [here](https://hypernews.cern.ch/HyperNews/CMS/get/tier0-Ops/1478.html)

**Comment by mtosi** [comment]
ciao
this is needed only for ZeroBias PD
=> only Prompt, no Express

the new PDs are populated only by a single HLT path
=> the AlCaRecoTriggerBit is not needed there

@the HIP task force this morning it has been decided that
those 2 PDs will stay until next MD (beginning of next week)
and then the 2 HLT paths will be added to ZeroBias PD

thanks much
  mia

**Comment by mmusich** [comment]
Hi Mia,

> this is needed only for ZeroBias PD => only Prompt, no Express 

OK fine

> the new PDs are populated only by a single HLT path => the AlCaRecoTriggerBit is not needed there 

sorry, I don't understand. We have only one Global Tag for every PD reconstructed in Prompt Reco. If I change it there, it will be changed for all. I assume this is harmless.

**Comment by mtosi** [comment]
ciao

On Wed, Aug 17, 2016 at 1:33 PM, Marco Musich <notifications@github.com>
wrote:

> Hi Mia,
>
> this is needed only for ZeroBias PD => only Prompt, no Express
>
> OK fine
>
> the new PDs are populated only by a single HLT path => the
> AlCaRecoTriggerBit is not needed there
>
> sorry, I don't understand. We have only one Global Tag for every PD
> reconstructed in Prompt Reco. If I change it there, it will be changed for
> all. I assume this is harmless.
>

of course, sorry my bad !
it is harmless, yes .. we make use of the new key/label in the
AlCaTriggerBit only on ZeroBias PD

thanks
  mia

**Comment by mmusich** [comment]
@mtosi @dmitrijus @vanbesien 
I am still puzzled. 
The tests of this PR show clearly that until https://github.com/cms-sw/cmssw/pull/15513 is merged we'll have this messages:
```
MSG-w GenericTriggerEventFlag:   TrackingMonitor:TrackerCollisionSelectedTrackMonZBnoHIPnoOOTgeneralTracks@streamBeginRun  12-Aug-2016 20:23:17 CEST Run: 165121
No logical expressions found under key Tracking_HLT_noHIP_noOOT in 'AlCaRecoTriggerBitsRcd'
%MSG
%MSG-w GenericTriggerEventFlag:   TrackingMonitor:TrackerCollisionSelectedTrackMonZBHIPnoOOTgeneralTracks@streamBeginRun  12-Aug-2016 20:23:17 CEST Run: 165121
No logical expressions found under key Tracking_HLT_HIP_noOOT in 'AlCaRecoTriggerBitsRcd'
%MSG
%MSG-w GenericTriggerEventFlag:   TrackingMonitor:TrackerCollisionSelectedTrackMonZBHIPOOTgeneralTracks@streamBeginRun  12-Aug-2016 20:23:17 CEST Run: 165121
No logical expressions found under key Tracking_HLT_HIP_OOT in 'AlCaRecoTriggerBitsRcd'
%MSG
```
in the [wf 1001.0 tests](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15449/14505/runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log).
Now, wf. 1001.0 runs the `DQM:@express matrix`:

```
cmsDriver.py step2  --conditions auto:run1_data -s RAW2DIGI,L1Reco,RECO,EI,ALCAPRODUCER:@allForExpress,DQM:@express,ENDJOB --process RECO --data  --eventcontent ALCARECO,DQM --runUnscheduled  --scenario pp --datatier ALCARECO,DQMIO --customise Configuration/DataProcessing/RecoTLR.customiseExpress -n 100  --filein filelist:step1_dasquery.log --lumiToProcess step1_lumiRanges.log --fileout file:step2.root  --suffix "-j JobReport2.xml "  > step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log  2>&1
```
but the claim is 

>     this is needed only for ZeroBias PD => only Prompt, no Express

This looks inconsistent to me. Please clarify

**Comment by mtosi** [comment]
ciao

I've just looked at the workflow

it is running
process.dqmoffline_step = cms.Path(process.DQMOfflineCommonSiStripZeroBias)
which is calling the TrackingDQMSourceTier0MinBias (which is meant to be run only on ZeroBias PD)

while @tier0
https://github.com/dmwm/T0/blob/master/etc/ProdOfflineConfiguration.py#L1301
no dqm_sequence is specified
=> I'm expecting the @common one is used
@dmitrijus @vanbesien , could you confirm it ?

=> on my opinion,
there is an inconsistency between what is run @ tier0 and this runTheMatrix workflow,
isn't there ?

**Comment by vanbesien** [comment]
Hi @mtosi ,

If no dqm_sequence is specified, it's in fact running all the sequences in both [DQMOffline_cff.py](https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DQMOffline/Configuration/python/DQMOffline_cff.py) and [DQMOffline_SecondStep_cff.py](https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DQMOffline/Configuration/python/DQMOffline_SecondStep_cff.py).
Meaning that specifying "@common" is even a subset of specifying nothing at all.

So at T0 for express basically everything is run for DQM.

**Comment by mtosi** [comment]
thanks much for the info
=> I would say that the workflow 1001.0 is not representing what tier0 runs
for Express
  mia

**Comment by mmusich** [comment]
@mtosi, from Broen's comment https://github.com/cms-sw/cmssw/pull/15449#issuecomment-240749654 I gather that everything is run on express, then an update of the Express GT is needed as well!

**Comment by mtosi** [comment]
https://github.com/cms-sw/cmssw/blob/CMSSW_8_1_X/DQMOffline/Configuration/python/autoDQM.py

@common
runs DQMOfflineCommon

@commonSiStripZeroBias
runs DQMOfflineCommonSiStripZeroBias

**Comment by mtosi** [comment]
thanks @mmusich 
I though this PR was already merged !
anyhow, I've just fixed the issue as it has been donw in https://github.com/cms-sw/cmssw/pull/15559/files

I hope this PR can be merged soon
before this fix only the sign by the ORP was missing ... even if the ORP happend yesterday ...

**Comment by Martin-Grunewald** [comment]
@mtosi 
please reduce the spurious commits included

**Comment by mtosi** [comment]
ciao
is it possible to merged this PR in the upcoming CMSSW release, please ?
thanks much
  mia

**Comment by mmusich** [comment]
please test

**Comment by dmitrijus** [comment]
+1

**Comment by mmusich** [comment]
as a reminder: to get consistent results https://github.com/cms-sw/cmssw/pull/15513 needs to be merged first

**Comment by mmusich** [comment]
please test

**Comment by mmusich** [comment]
@mtosi as expected after https://github.com/cms-sw/cmssw/pull/15513 has been merged, warnings disappear, please cf.

[old log](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15449/14705/runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log)

vs:
 
[new log](https://cmssdt.cern.ch/SDT/jenkins-artifacts/pull-request-integration/PR-15449/14816/runTheMatrix-results/1001.0_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5/step2_RunMinBias2011A+RunMinBias2011A+TIER0EXP+ALCAEXP+ALCAHARVD1+ALCAHARVD2+ALCAHARVD3+ALCAHARVD4+ALCAHARVD5.log)

@dmitrijus this PR is ready to be asked at the ORP today.

**Comment by mtosi** [comment]
thanks much
  mia

**Comment by davidlange6** [comment]
+1

**Comment by mtosi** [comment]
thanks !

**Pull request merged**